tests/include/test/drivers/hash.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Test driver for hash driver entry points.
+ */
+/*  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef PSA_CRYPTO_TEST_DRIVERS_HASH_H
+#define PSA_CRYPTO_TEST_DRIVERS_HASH_H
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "mbedtls/config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+
+#if defined(PSA_CRYPTO_DRIVER_TEST)
+#include <psa/crypto_driver_common.h>
+
+typedef struct {
+    /* If not PSA_SUCCESS, return this error code instead of processing the
+     * function call. */
+    psa_status_t forced_status;
+    /* Count the amount of times hash driver entry points are called. */
+    unsigned long hits;
+    /* Status returned by the last hash driver entry point call. */
+    psa_status_t driver_status;
+} test_driver_hash_hooks_t;
+
+#define TEST_DRIVER_HASH_INIT { 0, 0, 0 }
+static inline test_driver_hash_hooks_t test_driver_hash_hooks_init( void )
+{
+    const test_driver_hash_hooks_t v = TEST_DRIVER_HASH_INIT;
+    return( v );
+}
+
+extern test_driver_hash_hooks_t test_driver_hash_hooks;
+
+psa_status_t test_transparent_hash_compute(
```

Comment:
Even for testing only, I think we should avoid non-namespaced identifiers. They could conflict with the test framework of an embedded platform, for example. What't the motivation for not having a `mbedtls_` prefix?

I'd be ok with non-namespaced macros and types if they're internal to the library or tests, since they could only conflict with a header that our code includes. But names with external linkage can conflict with everything else that goes onto the platform.

Reply:
There is no special motivation for not having the `mbedtls_` prefix. I've just followed what we've been doing so far in the test drivers (see cipher.h, signature.h ...). I am going to add the prefix to all of them as I agree it will be better.

Reply:
I've addressed this now.